xline/src/storage/lease_store/mod.rs

```diff
@@ -277,11 +277,7 @@ impl LeaseStore {
 
     /// Keep alive a lease
     pub(crate) fn keep_alive(&self, lease_id: i64) -> Result<i64, ExecuteError> {
-        if !self.is_leader() {
-            return Err(ExecuteError::InvalidCommand(
-                "lease keep alive must be called on leader".to_owned(),
-            ));
-        }
+        assert!(self.is_leader(), "keep_alive must be called on leader");
```

Comment:
I guess it leadership change can happen before this function is called. If so, assert! will panic. So I think the previous solution is better.